wallet/src/ui/app/components/transactions-card/index.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+
+import AccountAddress from '_components/account-address';
+import ExplorerLink from '_components/explorer-link';
+import { ExplorerLinkType } from '_components/explorer-link/ExplorerLinkType';
+
+import type { ExecutionStatusType, TransactionKindName } from '@mysten/sui.js';
+
+import st from './TransactionsCard.module.scss';
+
+type TxnData = {
+    To?: string;
+    seq: number;
+    txId: string;
+    status: ExecutionStatusType;
+    txGas: number;
+    kind: TransactionKindName | undefined;
+    From: string;
+};
+const truncate = (txt: string, maxLength: number) => {
```

Comment:
Why not use the `useMiddleEllipsis` hook?

Reply:
`useMiddleEllipsis` Does not work inside the other function